app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt

```diff
@@ -534,19 +575,25 @@ class SettingsFragment : Fragment(R.layout.activity_settings_screen) {
 
     }
 
+    private fun isVpnLockdown(): Boolean? {
+        // lockdown vpn mode only on Q and above
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) return false
+        val vpnService = VpnController.getInstance().getBraveVpnService()
```

Comment:
use `Koin`.

Reply:
Created issue to move them to Koin. https://github.com/celzero/rethink-app/issues/303